plugins/source/gcp/client/client.go

```diff
@@ -173,6 +192,35 @@ func getProjectsV1(ctx context.Context, options ...option.ClientOption) ([]strin
 	return projects, nil
 }
 
+func getProjectsV1WithFilter(ctx context.Context, filter string, options ...option.ClientOption) ([]string, error) {
+	var (
+		projects []string
+	)
+	service, err := crmv1.NewService(ctx, options...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create cloudresourcemanager service: %w", err)
+	}
+
+	call := service.Projects.List().Filter(filter).Context(ctx)
+	for {
+		output, err := call.Do()
+		if err != nil {
+			return nil, err
+		}
+		for _, project := range output.Projects {
+			if project.LifecycleState == "ACTIVE" {
+				projects = append(projects, project.ProjectId)
+			}
+		}
```

Comment:
While I agree that this is a good protection for users, I think we need to explicitly warn users that we are skipping projects

Reply:
Before this PR, we just skipped them (because they make cloudquery hang).
I think it's out-of-scope to change this behaviour.

Reply:
I would argue that it isn't out of scope because before this we didn't expose the filter option. By doing this without telling the user our results will not match the results of a user just running `gcp projects -filter...`.